backend/src/main/java/org/example/backend/service/MovieService.java

```diff
@@ -48,4 +71,105 @@ public void deleteMovie(String slug) {
         movieRepo.deleteBySlug(slug);
     }
 
+    // fetch on search functionality
+
+    public List<Movie> fetchAndStoreMovies(String query) {
+        System.out.println("Starting movie fetching process...");
+
+        String netzkinoUrl = String.format("%s?q=%s&d=%s", NETZKINO_URL, query, netzkinoEnv);
+        System.out.println("Generated Netzkino URL: " + netzkinoUrl);
+
+        ResponseEntity<NetzkinoResponse> netzkinoResponse = restTemplate.getForEntity(netzkinoUrl, NetzkinoResponse.class);
+        NetzkinoResponse response = netzkinoResponse.getBody();
+
+        if (response == null || response.posts().isEmpty()) {
+            System.out.println("No movies found from API 1.");
+            return Collections.emptyList();
+        }
```

Comment:
`getBody()` can return `null`, causing a `NullPointerException` when calling `response.posts()`

```suggestion
        if (response == null || response.posts() == null || response.posts().isEmpty()) {           
            logger.warn("No movies found from API 1.");
            return Collections.emptyList();
        }
```